pkg/processor/build/builder.go

```diff
@@ -1552,6 +1564,71 @@ func (b *Builder) getS3FunctionItemKey() (string, error) {
 	return s3Attributes["s3ItemKey"], nil
 }
 
+func (b *Builder) resolveGitReference() (string, error) {
+	if b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitBranch"] != nil {
+		return fmt.Sprintf("refs/heads/%s", b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitBranch"]), nil
+	}
+	if b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitTag"] != nil {
+		return fmt.Sprintf("refs/tags/%s", b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitTag"]), nil
+	}
+	if b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitReference"] != nil {
+		return b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitReference"].(string), nil
+	}
+
+	return "", errors.New("No git reference was specified. (must specify gitBranch/gitTag/gitReference)")
+}
+
+func (b *Builder) parseFunctionGitCredentials() *githttp.BasicAuth {
+	var username, password string
+
+	_, found := b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitUsername"]
+	if found {
+		username = b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitUsername"].(string)
```

Comment:
hmm you're making an assumption the user won't try to provide his 123456789 password as an int? :)

I'd create a `resolveCodeEntryAttributeString(<attributes>) (string, error)` and call it for each field you need
it would get the value, assert on its type to be string and use it from here for both password and username (and branch, tag, reference, etc)